src/library/scala/sys/process/ProcessImpl.scala

```diff
@@ -147,6 +147,7 @@ private[process] trait ProcessImpl {
           throw err
         }
       runInterruptible {
+        source.join()
         val exit1 = first.exitValue()
```

Comment:
This was my comment in May. FSR github review made it a comment from now.

The answer to my question is `ThreadProcess` just sets up the stream for redirection. `PipedProcess` consumes the stream. `ThreadProcess` itself doesn't know if or when the i/o succeeded, alas.